### DISCUSSION TITLE
ERROR: [PostgreSQLClient:1252] The frontend is not connected to backend

### DISCUSSION THREAD

**Discussion by kevin-zue** [Marzin-bot/PostgreSQLClient#53] [Q&A]
Hi,

i am trying to use the PostgreSQL Client to connect to a PostgreSQL DB in Docker. My Docker Containers port is mapped to my local port:

![image](https://user-images.githubusercontent.com/118768421/204143637-a06fd898-eced-4af7-9b6f-36a1768989a4.png)

Im establishing a connection and get no error as response when i use the following script:

![image](https://user-images.githubusercontent.com/118768421/204143699-d8fddca2-ce8a-4bf4-8b7f-97f6762bc496.png)

But when i call the functions, that should actually execute requests, i get errors for every call of the execute function.

![image](https://user-images.githubusercontent.com/118768421/204143742-febaba0a-e63e-4fee-a4b4-939cbdbbe5a4.png)

![image](https://user-images.githubusercontent.com/118768421/204143744-462109bf-491c-402b-86cf-a05d4e4616a8.png)

Am i missing something?

Kind Regards,
Kevin

**Comment by KuwazyStudio** [comment]
Hello,
You must wait for the connection_established signal to be emitted before executing any requests.

Samuel Marzin

**Reply by kevin-zue** [reply]
Hi, 

thanks for your fast reply. I moved my execute call into the Function that is called on "connection_established". But the connection is never called, even after a couple of minutes.

Here i connect the functions and connect_to_host

![image](https://user-images.githubusercontent.com/118768421/204156693-1ebf7dac-1fe2-4dd5-8e1d-e40d27cbd967.png)

And in the connectionEstablished function i print a String

![image](https://user-images.githubusercontent.com/118768421/204156727-3ee1802e-3a79-4825-8784-a0c867a398b6.png)

This is the only print i get.

![image](https://user-images.githubusercontent.com/118768421/204156750-2cd633be-6468-4618-96e4-22d4ea17344b.png)


Kind Regards,
Kevin

**Reply by kevin-zue** [reply]
It seems like the Database is crashing and Restarting when i try to connect:

![image](https://user-images.githubusercontent.com/118768421/204158995-bee44638-f2ea-44d6-95e4-1e3ad14c1475.png)

**Reply by KuwazyStudio** [reply]
Consider trying to connect with another postgres client. If Postgres crashes then you can deduce that the problem is not with the Godot client.
